folly/CPortability.cpp

```diff
@@ -0,0 +1,563 @@
+
+#ifdef _MSC_VER
+#include <folly/CPortability.h>
+
+extern "C" {
+
+// Do nothing for the system log for now.
+// Do nothing for the system log for now.
+void openlog(const char*, int, int) {}
+void closelog() {}
+void syslog(int, const char*, ...) {}
+
+void __stdcall __alarm_callback_func(HWND, UINT, UINT_PTR, DWORD) {
+  raise(SIGALRM);
+}
+
+unsigned int alarm(unsigned int seconds) {
+  SetTimer(NULL, 0, (UINT)(seconds * 1000), __alarm_callback_func);
+  return 0;
+}
+
+char* asctime_r(const tm* tm, char* buf) {
+  char tmpBuf[64];
+  if (asctime_s(tmpBuf, sizeof(tmpBuf), tm))
+    return NULL;
+  return strcpy(buf, tmpBuf);
+}
+
+void bzero(void* s, size_t n) {
+  ZeroMemory(s, n);
+}
+
+int closedir(DIR* dir) {
+  if (!FindClose(dir->hand))
+    return -1;
+  free(dir);
+  return 0;
+}
+
+char* ctime_r(const time_t* t, char* buf) {
+  char tmpBuf[64];
+  if (ctime_s(tmpBuf, sizeof(tmpBuf), t))
+    return NULL;
+  return strcpy(buf, tmpBuf);
+}
+
+char* dirname(char* path) {
```

Comment:
Since this is Windows-only, perhaps you could annotate `path` as:

`_Inout_opt_z_`

...and the return value as:
`_Ret_z_`